submitter/relayer/relayer.go

```diff
@@ -386,44 +386,54 @@ func (rl *Relayer) buildTxWithData(
 	rl.logger.Debugf("Building a BTC tx using %v with data %x", utxo.TxID.String(), data)
 	tx := wire.NewMsgTx(wire.TxVersion)
 
-	outPoint := wire.NewOutPoint(utxo.TxID, utxo.Vout)
```

Comment:
the `utxo *types.UTXO` argument to the function is probably not needed now as all intput to the transaction as selected by `rl.BTCWallet.FundRawTransaction` function.

Given that we probably need two versions of this function:
1. first version for the first transaction, in which we allow wallet to chose inputs automatically
2. second version for the second transaction, in which we are able to specifiy input manually to use the transaction chaining

Reply:
Not sure how `FundRawTransaction` would simplify things as we cannot control the inputs used by `FundRawTransaction`. As such, for (1), how do we know the value of the output is sufficient to fund the second tx for chaining?

Reply:
If I understand our chaining mechanism correctly, want we want to do is:
1. send first transaction with pretty low fee rate, lets call it `Tx1`
2. then send the second transaction (`Tx2`)with higher fee rate and with input that is `Tx1` change output (this is necessary to achieve the chaining)

Imo `FundRawTransaction` can help here a lot and simplify stuff as:
1. For the `Tx1` we can just create unsigned and unfunded transaction with one output, and send it to `FundRawTransaction`, it will automatically choses inputs and add one change output which will have correct value to obey specified fee rate.
2. Then we can create `Tx2` with one input (change output from `Tx1` ) and one output and send it to `FundRawTransaction`. Now based on specified fee rate, endpoint will attach or not other inputs and will attach one change output to `Tx2` . This way chaining will be preserved,  and second tx will obey specified fee rate

Reply:
Oh, I didn't realize in (2) we can inject one input from the change output of tx1. If the case, then it indeed improves a lot. Sounds good to me!